### DISCUSSION TITLE
STM32F103C6T6 Cannot use UART on Pin PA9 PA10

### DISCUSSION THREAD

**Discussion by KKuKK** [stm32duino/Arduino_Core_STM32#1966] [Variants] [answered]
Hi 
I just bought a STM32F103C6T6 and tried to build a terminal for sended USB data. Works fine if I use 
the usb cable to the usb connection on the board. But I get no reaction form any UART pin on the board.
Not on Pin PA10, PA3 or PB6. 
I use Arduino 1.8.8 IDE with gerneric  STM32F103C6T6/fake STM32F103C8T6 and an tft 204x240 with ST7789.
All works fine on the mini usb port. I guess the UART are not enabled in the code. 
How can I use the RX pins on the board? 
Thanks for any help.
greetings Klaus

**Comment by JAndrassy** [comment]
I guess you use some USB to TTL Serial adapter. did you wire RX to TX and ground? how do you power the board? did you set USART support in Tools menu to enable it as Serial or Serial1?

**Comment by fpistm** [comment]
> with gerneric STM32F103C6T6/fake STM32F103C8T6

what you would mean by fake ?
How did you initialize UART? (code)?

**Comment by KKuKK** [comment, marked as answer]
Wau that is really great to get such a fast answer.I apologize, i just wanted to write the problem is solved. I was not fast enough.It’s exactly what you suggest. I don’t use an adapter, I just connect an arduino tx pin or an gps chip to the rx pin of the stm.Unfortunately there is no option in the toolsMenu for the stm32f103c6 for any usb settings, like for the c8t6. That’s why I thought there must be some declarations missing for my one. It’s quite confusing the different names and boards for the uart.I tried serial1 instead of serial. Now the usb connector isn’t working, but what I need is working. I want the RX pin PA10 as input,So I can use the terminal for debugging information from other chips, also for Gps Testing, etc.I am still confused. I thought PA10 is the rx pin for uart1, the same as the usb connector.Obviously wrong. Serial is the usb connector,Serial1 is pa10, pa9.My next test would be a small program to just check all the serial names and the mapping to the pins on the board.I first hoped I could use bin files for the C8t6 directly for the c6t6. No way, don’t work. This was stupid.Thanks for your help Greetings klausAm 08.03.2023 um 14:03 schrieb Juraj Andrássy ***@***.***>:﻿
I guess you use some USB to TTL Serial adapter. did you wire RX to TX and ground? how do you power the board? did you set USART support in Tools menu to enable it as Serial or Serial1?

**Reply by fpistm** [reply]
You should first read this:
https://github.com/stm32duino/Arduino_Core_STM32/wiki/API#hardwareserial

**Comment by KKuKK** [comment]
Fake meaning, there are wrong labels for the c8c6 out there from china. They are instead c6t6. That is the name in the tools menu for the stm32f103c6/fake stm32f103c8t6I choose this option to program my c6t6, which works.My one is a real stm. Checked it with cubeprogrammer from stm. After connecting you see all chip informations.Greetings klausVon meinem iPhone gesendetAm 08.03.2023 um 14:30 schrieb Frederic Pillon ***@***.***>:﻿

with gerneric STM32F103C6T6/fake STM32F103C8T6

what you would mean by fake ?
How did you initialize UART? (code)?

**Comment by KKuKK** [comment]
Ok thanks Von meinem iPhone gesendetAm 08.03.2023 um 14:45 schrieb Frederic Pillon ***@***.***>:﻿
You should first read this:
https://github.com/stm32duino/Arduino_Core_STM32/wiki/API#hardwareserial
